src/mono/mono/metadata/native-library.c

```diff
@@ -1223,11 +1223,32 @@ mono_loader_install_pinvoke_override (PInvokeOverrideFn override_fn)
 	pinvoke_override = override_fn;
 }
 
+static int mono_precompute_size (const char *key)
+{
+	int size = 1; // Null terminator
+	int len = (int)strlen (key);
+	for (int i = 0; i < len; ++i) {
+		unsigned char b = key[i];
+		if ((b >= '0' && b <= '9') ||
+		    (b >= 'a' && b <= 'z') ||
+			(b >= 'A' && b <= 'Z') ||
+			b == '_' || b == '.' || b == '-' || b ==  '+' || b == '<' || b == '>') {
```

Comment:
Make some helper functions:

```c
static gboolean
is_symbol_char_verbatim(char c)
{
	return ((b >= '0' && b <= '9') || (b >= 'a' && b <= 'z') || (b >= 'A' && 'b' <= 'Z');
}

static gboolean
is_symbol_char_underscore(char c)
{
	switch (c) {
	case '_':
	case '.':
	case '-':
	case '+':
	case '<':
	case '>':
		return TRUE;
	default:
		return FASLE;
	}
}
```

and use it here in `mono_precompute_size` and in `mono_fixup_symbol_name` so the logic is in one place.